tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -2053,6 +2064,98 @@ static __always_inline unsigned short get_inode_mode_from_fd(u64 fd)
     return READ_KERN(f_inode->i_mode);
 }
 
+static __always_inline void detect_hooked_syscalls(event_data_t *data){
+    int key = 0;
+    u64 *table_ptr = bpf_map_lookup_elem(&syscall_table_addr_map, (void *)&key);
+    if (table_ptr == NULL)
+        return ;
+
+    u64 *start_ptr = bpf_map_lookup_elem(&code_boundaries_map, (void *)&key);
+    if (start_ptr == NULL)
+        return ;
+
+    key++;
+    u64 *end_ptr= bpf_map_lookup_elem(&code_boundaries_map, (void *)&key);
+    if (end_ptr == NULL)
+        return ;
+
+    unsigned long *syscall_table_addr = (unsigned long*) *table_ptr;
+    unsigned long stext = *start_ptr;
+    unsigned long etext = *end_ptr;
+
+    #if defined(bpf_target_x86)
+        int syscalls_to_check[] = {
+            0,          // read
+            1,          // write
+            2,          // open
+            3,          // close
+            16,         // ioctl
+            41,         // socket
+            59,         // execve
+            62,         // kill
+            78,         // getdents
+            101,        // ptrace
+            217,        // getdents64
+            321,        // bpf
+            322,        // execveat
+            322,        // openat
+            44,          // sendto
+            45,         // recvfrom
+            46,     	// sendmsg
+            47	        // recvmsg
+
+        };
+    #elif defined(bpf_target_arm64)
+        int syscalls_to_check[] = {
+            63,         // read
+            64,         // write
+                        // open (not present)
+            57,         // close
+            29,         // ioctl
+            198,        // socket
+            221,        // execve
+            129,        // kill
+                        // getdents (not present)
+            117,        // ptrace
+            61,         // getdents64
+            280,        // bpf
+            281,        // execveat
+            56,         // openat
+            206,        // sendto
+            207,        // recvfrom
+            211,        // sendmsg
+            212	        // recvmsg
+
+        };
```

Comment:
This information can be configured from userspace (using a bpf map) instead of being hardcoded.
This will allow to extend this feature to other syscalls dynamically.